api/organisations/views.py

```diff
@@ -241,13 +244,16 @@ def permissions(self, request):
     @action(detail=True, methods=["GET"], url_path="my-permissions")
     def my_permissions(self, request, pk):
         org = self.get_object()
-        permission_keys = request.user.get_permission_keys_for_organisation(org)
-        serializer = self.get_serializer(
-            instance={
-                "permissions": permission_keys,
-                "admin": request.user.is_organisation_admin(org),
-            }
+
+        organisation_permissions_calculator = OrganisationPermissionsCalculator(
+            pk=org.id
+        )
+        permission_data = organisation_permissions_calculator.get_permission_data(
+            user_id=request.user.id,
+            is_organisation_admin=request.user.is_organisation_admin(org),
```

Comment:
Yeah, this is sort of a weird one since you can be a project/environment admin using the usual `UserPermissionGroupProjectPermission` or `UserProjectPermission` but you can be organisation permission only using `UserOrganisation`, it will make more sense if you look at the refactored permisson_calculator now I think